src/middlewared/middlewared/plugins/auth.py

```diff
@@ -325,9 +349,130 @@ def token(self, app, token):
         return True
 
 
+class TwoFactorAuthService(ConfigService):
+
+    class Config:
+        datastore = 'system.twofactorauthentication'
+        datastore_extend = 'auth.twofactor.two_factor_extend'
+        namespace = 'auth.twofactor'
+
+    @private
+    async def two_factor_extend(self, data):
+        data['secret'] = await self.middleware.call('pwenc.decrypt', data['secret'])
+
+        for srv in ['ssh']:
+            data['services'].setdefault(srv, False)
+
+        return data
+
+    @accepts(
+        Dict(
+            'auth_twofactor_update',
+            Bool('enabled'),
+            Int('otp_digits', validators=Range(min=6, max=8)),
+            Int('window', validators=Range(min=0)),
+            Int('interval', validators=Range(min=5)),
+            Dict(
+                'services',
+                Bool('ssh', default=False)
+            ),
+            update=True
+        )
+    )
+    async def do_update(self, data):
+        """
+        `otp_digits` represents number of allowed digits in the OTP.
+
+        `window` extends the validity to `window` many counter ticks before and after the current one.
```

Comment:
I'm confused about this sentence. Is the second "\`window\`" a verb in this case?